cmd/oras/root/manifest/index/update.go

```diff
@@ -0,0 +1,235 @@
+	"fmt"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras/cmd/oras/internal/argument"
+	"oras.land/oras/cmd/oras/internal/command"
+	"oras.land/oras/cmd/oras/internal/display/status"
+	oerrors "oras.land/oras/cmd/oras/internal/errors"
+	"oras.land/oras/cmd/oras/internal/option"
+	"oras.land/oras/cmd/oras/internal/output"
+	"oras.land/oras/internal/contentutil"
+	"oras.land/oras/internal/descriptor"
+)
+
+type updateOptions struct {
+	option.Common
+	option.Target
+
+	addArguments    []string
+	mergeArguments  []string
+	removeArguments []string
+	tags            []string
+}
+
+func updateCmd() *cobra.Command {
+	var opts updateOptions
+	cmd := &cobra.Command{
+		Use:   "update <name>{:<tag>|@<digest>} {--add/--merge/--remove} {<tag>|<digest>} [...]",
+		Short: "[Experimental] Update and push an image index",
+		Long: `[Experimental] Update and push an image index. All manifests should be in the same repository
+		
+Example - remove a manifest and add two manifests from an index tagged 'v1'. The tag will point to the updated index:
+  oras manifest index update localhost:5000/hello:v1 --add linux-amd64 --add linux-arm64 --remove sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+
+Example - update an index by specifying its digest:
+  oras manifest index update localhost:5000/hello@sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9 --add linux-amd64 --remove sha256:fd6ed2f36b5465244d5dc86cb4e7df0ab8a9d24adc57825099f522fe009a22bb
+
+Example - merge manifests from the index 'v2-windows' to the index 'v2':
+  oras manifest index update localhost:5000/hello:v2 --merge v2-windows
+
+Example - update an index and tag the updated index as 'v2.1.0' and 'v2':
+  oras manifest index update localhost:5000/hello@sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9 --add linux-amd64 --tag "v2.1.0" --tag "v2"
+  `,
+		Args: oerrors.CheckArgs(argument.Exactly(1), "the destination index to update"),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.RawReference = args[0]
+			return option.Parse(cmd, &opts)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return updateIndex(cmd, opts)
+		},
+	}
+	option.ApplyFlags(&opts, cmd.Flags())
+	cmd.Flags().StringArrayVarP(&opts.addArguments, "add", "", nil, "manifests to add to the index")
+	cmd.Flags().StringArrayVarP(&opts.mergeArguments, "merge", "", nil, "indexes to be merged into the index")
+	cmd.Flags().StringArrayVarP(&opts.removeArguments, "remove", "", nil, "manifests to remove from the index, must be digests")
+	cmd.Flags().StringArrayVarP(&opts.tags, "tag", "", nil, "extra tags for the updated index")
+	return oerrors.Command(cmd, &opts.Target)
+}
+
+func updateIndex(cmd *cobra.Command, opts updateOptions) error {
+	// if no update flag is used, do nothing
```

Comment:
If no update flag is used, should we return an error?

Reply:
I don't think returning an error is necessary. The command of updating nothing shouldn't fail, informing the user that nothing is done should be enough.

Reply:
It is interesting that if there are no updates,
- `usermod` returns `2` with an error.
- `groupmod` passes with `0` returned.

Reply:
So.. I think either is fine.